### ISSUE TITLE
Ensure Content-Type is the same irrespective of whether asset is served from NFS or S3

### ISSUE THREAD

**Issue by chrisroos** [alphagov/asset-manager#175]
While testing issue #149 I noticed that the value of the `Content-Type` header was different when the file was served from NFS (`application/octet-stream`) vs when it was served from S3 (`*/*; charset=utf-8`).

We should ensure the value of the `Content-Type` header for the [commonly uploaded asset filetypes][asset-filetypes] is the identical when served from NFS or S3. It might be that the problem I saw in issue #149 was due to me using an unusual file extension (`.dat`) and that for everything else it might work just fine.

[asset-filetypes]: https://github.com/alphagov/asset-manager/blob/master/docs/existing_assets.md#file-extensions

**Comment by chrisroos** [comment]
I've investigated the `Content-Type` being reported for each of [these 20 file extensions](https://github.com/alphagov/asset-manager/blob/master/docs/existing_assets.md#file-extensions).

In each case I created an empty file with the relevant file extension using `touch file.<extension>`. I uploaded the file to my local asset-manager and then requested it from the filesystem and from S3. I did this testing in the Rails development environment but also tested a couple of them in the Rails production environment to ensure we were seeing the same results.

| File extension | Filesystem               | S3                 |
| -------------- | ------------------------ | ------------------ |
| pdf            | application/pdf          | application/pdf    |
| jpg            | image/jpeg               | image/jpeg         |
| doc            | application/octet-stream | \*/*; charset=utf-8 |
| png            | image/png                | image/png          |
| gif            | image/gif                | image/gif          |
| docx           | application/octet-stream | \*/*; charset=utf-8 |
| xlsx           | application/octet-stream | \*/*; charset=utf-8 |
| xls            | application/octet-stream | \*/*; charset=utf-8 |
| odt            | application/octet-stream | \*/*; charset=utf-8 |
| ppt            | application/octet-stream | \*/*; charset=utf-8 |
| zip            | application/zip          | application/zip    |
| ods            | application/octet-stream | \*/*; charset=utf-8 |
| pptx           | application/octet-stream | \*/*; charset=utf-8 |
| xlsm           | application/octet-stream | \*/*; charset=utf-8 |
| url            | application/octet-stream | \*/*; charset=utf-8 |
| csv            | text/csv                 | text/csv           |
| jpeg           | image/jpeg               | image/jpeg         |
| pd             | application/octet-stream | \*/*; charset=utf-8 |
| txt            | text/plain               | text/plain         |
| webp           | application/octet-stream | \*/*; charset=utf-8 |

**Comment by chrisroos** [comment]
It looks as though the difference is in the Rails app (rather than in the nginx config):

```
$ curl http://localhost:3037/media/59a81844759b74425ab3ccca/file.webp -v
< HTTP/1.1 200 OK 
< Content-Type: application/octet-stream

$ curl http://localhost:3037/media/59a81844759b74425ab3ccca/file.webp?proxy_to_s3_via_nginx=true -v
< HTTP/1.1 200 OK 
< Content-Type: */*; charset=utf-8
```

**Comment by chrisroos** [comment]
I've created PR #177 that contains a fix for this issue.

**Comment by chrisroos** [comment]
I've merged PR #177 and plan to test the behaviour in integration once it's been deployed.

**Comment by chrisroos** [comment]
I've just checked this in integration:

```
$ echo `date` > tmp.doc

$ curl https://asset-manager.integration.publishing.service.gov.uk/assets --form "asset[file]=@tmp.doc" -H"Authorization: Bearer `cat /etc/govuk/manuals-publisher/env.d/ASSET_MANAGER_BEARER_TOKEN`" -H"Accept: application/json" 
{"_response_info":{"status":"created"},"id":"https://asset-manager.integration.publishing.service.gov.uk/assets/59ad3f1ced915d6253709d18","name":"tmp.doc","content_type":"application/msword","file_url":"https://assets-origin.integration.publishing.service.gov.uk/media/59ad3f1ced915d6253709d18/tmp.doc","state":"unscanned"}

$ curl https://assets-origin.integration.publishing.service.gov.uk/media/59ad3f1ced915d6253709d18/tmp.doc -v
< HTTP/1.1 200 OK
< Content-Type: application/octet-stream

$ curl https://assets-origin.integration.publishing.service.gov.uk/media/59ad3f1ced915d6253709d18/tmp.doc?proxy_to_s3_via_nginx=true -v
< HTTP/1.1 200 OK
< Content-Type: application/octet-stream
```

I'm happy that it's working as expected so am closing this issue.
